### PR TITLE
docs: terraform module jinad-aws

### PR DESCRIPTION
**Pull request by alanthssss**
### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [ ] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [x] Are your section headers in sentence case?

### CODE REVIEW

**Comment by alanthssss** [comment]
> releated issue: https://github.com/jina-ai/jina/issues/2087

It would be very nice for me if you reviewers give some advices about the format/content in this PR. 
Thank you all!

File: chapters/remote/jinad.md [hunk 1]
```diff
@@ -16,6 +16,16 @@ We use the tag `latest-daemon` in the below command, which uses the latest relea
 docker pull jinaai/jina:latest-daemon
 ```
 
+##### Deploying AWS EC2 instance with inside jinad by [terraform](https://www.terraform.io/).
+
+We extract a module [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) to simplify this work. 
+
+By this terraform module. We can:
```
Comment (CatStark) [inline chapters/remote/jinad.md]: With this terraform module we can:

File: chapters/remote/jinad.md [hunk 2]
```diff
@@ -16,6 +16,16 @@ We use the tag `latest-daemon` in the below command, which uses the latest relea
 docker pull jinaai/jina:latest-daemon
 ```
 
+##### Deploying AWS EC2 instance with inside jinad by [terraform](https://www.terraform.io/).
+
+We extract a module [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) to simplify this work. 
+
+By this terraform module. We can: 
+- Deploy one or more instance with inside jinad
```
Comment (CatStark) [inline chapters/remote/jinad.md]: Remove "with" please

File: chapters/remote/jinad.md [hunk 3]
```diff
@@ -16,6 +16,16 @@ We use the tag `latest-daemon` in the below command, which uses the latest relea
...
+- Custom the type and size of the root disk
+- Add some other python packages as required
```
Comment (CatStark) [inline chapters/remote/jinad.md]: Remove "some" please

File: chapters/remote/jinad.md [hunk 4]
```diff
@@ -16,6 +16,16 @@ We use the tag `latest-daemon` in the below command, which uses the latest relea
 docker pull jinaai/jina:latest-daemon
 ```
 
+##### Deploying AWS EC2 instance with inside jinad by [terraform](https://www.terraform.io/).
```
Comment (CatStark) [inline chapters/remote/jinad.md]: Remove "with" please

**Review by CatStark — DISMISSED** [review]
Just some changes :)

**Comment by CatStark** [comment]
Could you maybe try to be more in depth on the steps? Following the template should also help to see how to structure it

**Comment by alanthssss** [comment]
> Could you maybe try to be more in depth on the steps? Following the template should also help to see how to structure it

Sure.

**Review by FionnD — CHANGES_REQUESTED** [review]
Hi @niuzs-alan 

Thanks!

Unfortunately, we need a lot more details. We need a step by step guide on how a user can deploy Jina via terraform. 

Please use this [template](https://github.com/jina-ai/docs/blob/master/page_templates/developer_guide_how_to.rst) 

We need something similar to 
1. https://jina.ai/2020/11/05/Terraform-plus-Jina.html
2. https://rasa.com/docs/rasa-x/installation-and-setup/install/helm-chart/

**Comment by alanthssss** [comment]
> Hi @niuzs-alan
> 
> Thanks!
> 
> Unfortunately, we need a lot more details. We need a step by step guide on how a user can deploy Jina via terraform.
> 
> Please use this [template](https://github.com/jina-ai/docs/blob/master/page_templates/developer_guide_how_to.rst)
> 
> We need something similar to
> 
> 1. https://jina.ai/2020/11/05/Terraform-plus-Jina.html
> 2. https://rasa.com/docs/rasa-x/installation-and-setup/install/helm-chart/

Thanks for the request,  @FionnD .

Here is a clear usage in [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) referring to its source code's README.md.
Do we need further details for this practice ?

As for the template, I will read it carefully and follow it.

**Comment by FionnD** [comment]
> > Hi @niuzs-alan
> > Thanks!
> > Unfortunately, we need a lot more details. We need a step by step guide on how a user can deploy Jina via terraform.
> > Please use this [template](https://github.com/jina-ai/docs/blob/master/page_templates/developer_guide_how_to.rst)
> > We need something similar to
> > 
> > 1. https://jina.ai/2020/11/05/Terraform-plus-Jina.html
> > 2. https://rasa.com/docs/rasa-x/installation-and-setup/install/helm-chart/
> 
> Thanks for the request, @FionnD .
> 
> Here is a clear usage in [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) referring to its source code's README.md.
> Do we need further details for this practice ?
> 
> As for the template, I will read it carefully and follow it.

Hi @niuzs-alan 

Very good question! unfortunately yes we do need further details. unfortunately, our users may not be experienced with using terraform before and they need to know how to start :). Does that make sense?

**Comment by alanthssss** [comment]
> > > Hi @niuzs-alan
> > > Thanks!
> > > Unfortunately, we need a lot more details. We need a step by step guide on how a user can deploy Jina via terraform.
> > > Please use this [template](https://github.com/jina-ai/docs/blob/master/page_templates/developer_guide_how_to.rst)
> > > We need something similar to
> > > 
> > > 1. https://jina.ai/2020/11/05/Terraform-plus-Jina.html
> > > 2. https://rasa.com/docs/rasa-x/installation-and-setup/install/helm-chart/
> > 
> > 
> > Thanks for the request, @FionnD .
> > Here is a clear usage in [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) referring to its source code's README.md.
> > Do we need further details for this practice ?
> > As for the template, I will read it carefully and follow it.
> 
> Hi @niuzs-alan
> 
> Very good question! unfortunately yes we do need further details. unfortunately, our users may not be experienced with using terraform before and they need to know how to start :). Does that make sense?

Cool, I got the inspiration.
I would like to add more details as required later.

**Comment by FionnD** [comment]
Hi @niuzs-alan 

Sorry for not being to answer during the meeting! What was the question?

File: chapters/remote/jinad.md [hunk 5]
```diff
@@ -16,6 +16,16 @@ We use the tag `latest-daemon` in the below command, which uses the latest relea
 docker pull jinaai/jina:latest-daemon
 ```
 
+##### Deploying AWS EC2 instance inside jinad by [terraform](https://www.terraform.io/)
+
+We extract a module [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) to simplify this work. 
+
+With this terraform module We can: 
+- Deploy one or more instance inside jinad
```
Comment (nan-wang) [inline chapters/remote/jinad.md]: Be careful of using the same word `instance`. The reader will get confused between EC2 instance and a jina instance.
```suggestion
- Deploy one or more Flows inside jinad
```

File: chapters/remote/jinad.md [hunk 6]
```diff
@@ -16,6 +16,16 @@ We use the tag `latest-daemon` in the below command, which uses the latest relea
 docker pull jinaai/jina:latest-daemon
 ```
 
+##### Deploying AWS EC2 instance inside jinad by [terraform](https://www.terraform.io/)
+
+We extract a module [jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) to simplify this work. 
+
+With this terraform module We can:
```
Comment (nan-wang) [inline chapters/remote/jinad.md]:
```suggestion
With this terraform module we can: 
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Comment by nan-wang** [comment]
@niuzs-alan Make sure you have read through the documentation style guide. https://github.com/jina-ai/docs/blob/master/page_templates/style_guide.md

File: chapters/remote/jinad.md (hunk 5 above)
Comment (alanthssss) [reply chapters/remote/jinad.md]: Cool, that's the point.
Let me make this sentence more clear.

**Comment by deepankarm** [comment]
@niuzs-alan Here's what we need for the docs.

Add a page under [Distributed Jina](https://docs.jina.ai/chapters/remote/) named `Deploying JinaD on AWS with Terraform` which should have the following sections:

- Mention what Terraform is and how to install it. 
- Detailed explanation of the input/output args for the module.
- Step by step (runnable) commands to use our module
   - Configure AWS credentials on local (make sure not to commit any secrets to the docs) 
   - terraform init
   - terraform deploy
   - How to use the IP:PORT once the instances are deployed
   - How to debug if there are failures during deployment
   - terraform destroy

Even though the official module in terraform registry is well-documented, docs.jina.ai would be entrypoint for Jina users.

**Comment by alanthssss** [comment]
> @niuzs-alan Here's what we need for the docs.
> 
> Add a page under [Distributed Jina](https://docs.jina.ai/chapters/remote/) named `Deploying JinaD on AWS with Terraform` which should have the following sections:
> 
> * Mention what Terraform is and how to install it.
> * Detailed explanation of the input/output args for the module.
> * Step by step (runnable) commands to use our module
>   
>   * Configure AWS credentials on local (make sure not to commit any secrets to the docs)
>   * terraform init
>   * terraform deploy
>   * How to use the IP:PORT once the instances are deployed
>   * How to debug if there are failures during deployment
>   * terraform destroy
> 
> Even though the official module in terraform registry is well-documented, docs.jina.ai would be entrypoint for Jina users.

Cool! Thank you for your recommendation.
Let me add the above details!

**Comment by niuzs-alan** [comment]
https://github.com/jina-ai/jina/issues/2087

**Comment by alanthssss** [comment]
@FionnD Hi, Fionn, I did not follow our [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) strictly.
Could you plz check the structure of the updates? 
And should I follow it strictly? If so, I'd like to follow the templates.

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
+# Deploying JinaD on AWS With Terraform
+
+## Overview
+
+Here is a [terraform module jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) to deploy `JinaD` on AWS by terraform.
```
Comment (FionnD) [inline chapters/remote/deploy-jinad-on-aws-with-terraform.md]: The Jina AI team has released a [terraform module jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest) to deploy `JinaD` on AWS by terraform.

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+With this terraform module you can:
+- Custom the type and size of the root disk
```
Comment (FionnD) [inline chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Customise the type and size of the root disk

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+- Deploy multiple AWS EC2 instance
+- Execute extra command line
```
Comment (FionnD) [inline chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Remove this point

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+### [Terraform CLI](https://www.terraform.io/docs/cli/index.html)
+
+- Basic 
+   1. `terraform init` is used to initialize a working directory containing Terraform configuration files
+   2. `terraform apply` executes the actions proposed in a Terraform plan
+   3. `terraform output` extract the value of an output variable from the state file
+   4. `terraform destory` is a convenient way to destroy all remote objects managed by a particular Terraform configuration
+- More
```
Comment (FionnD) [inline chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Advanced instead of More

**Review by FionnD — DISMISSED** [review]
Hi @niuzs-alan 

Super work! I left some small comments. 

If @deepankarm is happy with it, good to merge

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 9 above)
Comment (alanthssss) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: uh-ha, which point?

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 7 above)
Comment (alanthssss) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: And I should remove the `a` before the [terraform module jinad-aws](https://registry.terraform.io/modules/jina-ai/jinad-aws/jina/latest)

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 10 above)
Comment (alanthssss) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Actually more appropriate.

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 7 above)
Comment (alanthssss) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Done.

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 8 above)
Comment (alanthssss) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Done.

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 10 above)
Comment (alanthssss) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Done.

**Review by deepankarm — APPROVED** [review]
@niuzs-alan great work! Content looks good to me.

**Comment by alanthssss** [comment]
> @niuzs-alan great work! Content looks good to me.

@deepankarm  Thanks for your clear guidance and brilliant pre-works!

**Comment by alanthssss** [comment]
> Hi @niuzs-alan
> 
> Sorry for not being to answer during the meeting! What was the question?

Thank you! 
No other questions.
Just sync the process, haha.

File: chapters/remote/deploy-jinad-on-aws-with-terraform.md (hunk 9 above)
Comment (FionnD) [reply chapters/remote/deploy-jinad-on-aws-with-terraform.md]: Ha good question, 

`Execute extra command line`

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**